docs/pages/reference/backends.mdx

```diff
@@ -172,6 +172,143 @@ teleport:
+  connect directly to the the auth service have the `auth_server` configuration
+  setting populated with the address of a load balancer for auth service
+  instances.
+
+```yaml
+teleport:
+  storage:
+    type: postgresql
+
+    # conn_string is a libpq-compatible connection string (see
+    # https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING);
+    # pool_max_conns is an additional parameter that determines the maximum
+    # number of connections in the connection pool used for the cluster state
+    # database (the change feed uses an additional connection), defaulting to
+    # a value that depends on the number of available CPUs.
+    conn_string: postgresql://user_name@database-address/teleport_backend?sslmode=verify-full&pool_max_conns=20
+
+    # An audit_events_uri with a scheme of postgresql:// will use the
+    # PostgreSQL backend for audit log storage; the URI is a libpq-compatible
+    # connection string just like the cluster state conn_string, but cannot be
+    # specified as key=value pairs. It's possible to specify completely
+    # different PostgreSQL clusters for cluster state and audit log.
+    audit_events_uri:
+      - postgresql://user_name@database-address/teleport_audit?sslmode=verify-full
+```
+
+Audit log events are periodically deleted after a default retention period of
+3766 hours (one year); it's possible to select a different retention period or
+to disable the cleanup entirely, by specifying the `retention_period` or the
+`disable_cleanup` parameters in the fragment of the URI:
+
+```yaml
+storage:
+  teleport:
+    audit_events_uri:
+      - postgresql://user_name@database-address/teleport_audit?sslmode=verify-full#disable_cleanup=false&retention_period=2160h
+```
+
+### Authentication
+
+We strongly recommend using [client
+certificates](https://www.postgresql.org/docs/current/ssl-tcp.html#SSL-CLIENT-CERTIFICATES)
+to authenticate Teleport to PostgreSQL, as well as enforcing the use of TLS and
+[verifying the server
+certificate](https://www.postgresql.org/docs/current/libpq-ssl.html#LIBQ-SSL-CERTIFICATES)
+on the client side. If the use of passwords is unavoidable, we recommend
+configuring them in [the `~/.pgpass`
+file](https://www.postgresql.org/docs/current/libpq-pgpass.html) rather than
+storing them in the configuration file.
+
+### Azure AD authentication
+
+If you are running Teleport on Azure, Teleport can make use of [Azure AD
+authentication](https://learn.microsoft.com/azure/postgresql/flexible-server/concepts-azure-ad-authentication)
+to connect to an [Azure Database for
+PostgreSQL](https://azure.microsoft.com/en-us/products/postgresql) server
+without having to manage any secrets:
+
+```yaml
+teleport:
+  storage:
+    type: postgresql
+
+    conn_string: postgresql://user_name@database-name.postgres.database.azure.com/teleport_backend?sslmode=verify-full&pool_max_conns=20
+    auth_mode: azure
```

Comment:
Can we default to azure mode for Azure databases? Or even just mandate Azure IAM for them?

Reply:
In one of the RFDs we agreed that we don't want to special-case the `.postgres.database.azure.com` suffix in case they change it in the future (maybe for some variation of their cloud?).

The chart will have a dedicated `azure` configuration mode, FWIW.